src/node_process_object.cc

```diff
@@ -105,18 +105,42 @@ MaybeLocal<Object> CreateProcessObject(Realm* realm) {
   READONLY_PROPERTY(
       process, "version", FIXED_ONE_BYTE_STRING(isolate, NODE_VERSION));
 
-  // process.versions
   Local<Object> versions = Object::New(isolate);
-  READONLY_PROPERTY(process, "versions", versions);
+  // Node.js version is always on the top
+  READONLY_STRING_PROPERTY(
+      versions, "node", per_process::metadata.versions.node);
+
+#define V(key) +1
+  std::pair<std::string_view, std::string_view>
+      versions_array[NODE_VERSIONS_KEYS(V)];
+#undef V
+  auto* slot = &versions_array[0];
 
 #define V(key)                                                                 \
-  if (!per_process::metadata.versions.key.empty()) {                           \
-    READONLY_STRING_PROPERTY(                                                  \
-        versions, #key, per_process::metadata.versions.key);                   \
-  }
+  do {                                                                         \
+    *slot++ = std::make_pair(#key, per_process::metadata.versions.key);        \
```

Comment:
Fix for `process.versions`:
```suggestion
    *slot++ = std::make_pair<std::string_view, std::string_view>(              \
        #key, per_process::metadata.versions.key);                             \
```
(same reasoning as https://github.com/nodejs/node/pull/46428#discussion_r1095943538 - if you don't mention the types explicitly, it's gonna infer `std::string` as the type for the second param because `per_process::metadata.versions.key` is an `std::string`, so it creates a `std::string_view` from a temporary `std::string` during assignment, which causes UB)

Reply:
That's a good point, actually. I'd forgotten they're std::string objects.

Tangentially: that's kind of bad because it introduces an avoidable heap allocation. `process.v8.versions` is approx. 19 bytes and that's larger than the SSO (short string optimization) threshold on some systems.

It'd be better to declare the version strings as `char key[24]` or `char key[32]` rather than `std::string key` in src/node_metadata.h.